src/NServiceBus.Core/MessageMutators/Mutators.cs

```diff
@@ -9,11 +12,21 @@ public Mutators()
 
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.Pipeline.Register("MutateIncomingTransportMessage", new MutateIncomingTransportMessageBehavior(), "Executes IMutateIncomingTransportMessages");
-            context.Pipeline.Register("MutateIncomingMessages", new MutateIncomingMessageBehavior(), "Executes IMutateIncomingMessages");
+            var registry = context.Settings.GetOrDefault<RegisteredMutators>() ?? new RegisteredMutators();
+
+            context.Pipeline.Register("MutateIncomingTransportMessage", b => new MutateIncomingTransportMessageBehavior(registry.IncomingTransportMessage), "Executes IMutateIncomingTransportMessages");
```

Comment:
Why did this one entry change to the `Func` overload?

Reply:
Misstake, we contemplated to pass conatiner mutators in but realized that we had to resolve them on each call to preserve behavior.

Fixed now